ibis/expr/schema.py

```diff
@@ -275,49 +291,63 @@ def apply_to(self, df: pd.DataFrame) -> pd.DataFrame:
         return df
 
 
-schema = Dispatcher('schema')
-infer = Dispatcher('infer')
+@lazy_singledispatch
+def schema(value: Any) -> Schema:
+    """Construct ibis schema from schema-like python objects."""
+    raise InputTypeError(value)
 
 
-@schema.register()
-def schema_from_kwargs(**kwargs):
-    return Schema(kwargs)
+@lazy_singledispatch
+def infer(value: Any, schema=None) -> Schema:
+    """Infer the corresponding ibis schema for a python object."""
+    raise InputTypeError(value)
 
 
 @schema.register(Schema)
-def schema_from_schema(s):
+def from_schema(s):
     return s
 
 
 @schema.register(Mapping)
-def schema_from_mapping(d):
+def from_mapping(d):
     return Schema(d)
 
 
 @schema.register(Iterable)
-def schema_from_pairs(lst):
+def from_pairs(lst):
     return Schema.from_tuples(lst)
 
 
 @schema.register(type)
-def schema_from_class(cls):
+def from_class(cls):
     return Schema(dt.dtype(cls))
 
 
-@schema.register(Iterable, Iterable)
-def schema_from_names_types(names, types):
```

Comment:
Removed this overload so that we can use lazy singledispatcher in order to eliminate mandatory pandas dependency later on. Hence I moved this logic to `Schema.from_tuples()`